source/utilities.cc

```diff
@@ -2363,6 +2364,84 @@ namespace aspect
         }
     }
 
+    Operator::Operator()
+      :
+      op(uninitialized)
+    {}
+
+
+    Operator::Operator(const operation _op)
+      :
+      op(_op)
+    {}
+
+
+    double
+    Operator::operator() (double x, double y) const
+    {
+      switch (op)
+        {
+          case Utilities::Operator::add:
+          {
+            return x + y;
+            break;
+          }
+          case Utilities::Operator::subtract:
+          {
+            return x - y;
+            break;
+          }
+          case Utilities::Operator::minimum:
+          {
+            return std::min(x,y);
+            break;
+          }
+          case Utilities::Operator::maximum:
+          {
+            return std::max(x,y);
+            break;
+          }
+          default:
+          {
+            Assert ( false, ExcMessage ("ASPECT's operator class only supports the following operators: "
+                                        "add, subtract, minimum and maximum. Please check your parameter file.") );
```

Comment:
But that's a programming error, not something that has to do with user input files. So I'd still favor `ExcInternalError()`.